crates/sncast/src/starknet_commands/account/create.rs

```diff
@@ -7,22 +7,29 @@ use camino::Utf8PathBuf;
 use clap::Args;
 use serde_json::json;
 use sncast::helpers::configuration::CastConfig;
-use sncast::helpers::constants::{CREATE_KEYSTORE_PASSWORD_ENV_VAR, OZ_CLASS_HASH};
+use sncast::helpers::constants::{
+    ARGENT_CLASS_HASH, CREATE_KEYSTORE_PASSWORD_ENV_VAR, OZ_CLASS_HASH,
+};
 use sncast::response::structs::{AccountCreateResponse, Felt};
 use sncast::{
     check_class_hash_exists, check_if_legacy_contract, extract_or_generate_salt, get_chain_id,
     get_keystore_password, handle_account_factory_error, parse_number,
 };
-use starknet::accounts::{AccountFactory, OpenZeppelinAccountFactory};
+use starknet::accounts::{
+    AccountDeployment, AccountFactory, ArgentAccountFactory, OpenZeppelinAccountFactory,
+};
 use starknet::core::types::{FeeEstimate, FieldElement};
-use starknet::core::utils::get_contract_address;
 use starknet::providers::jsonrpc::HttpTransport;
 use starknet::providers::JsonRpcClient;
 use starknet::signers::{LocalWallet, SigningKey};
 
 #[derive(Args, Debug)]
 #[command(about = "Create an account with all important secrets")]
 pub struct Create {
+    /// Type of the account
+    #[clap(value_enum, short = 't', long = "type", default_value_t = AccountType::Oz)]
```

Comment:
I think this should be required for custom class hashes, otherwise we have no way to know the account type.